src/toil/fileStores/abstractFileStore.py

```diff
@@ -430,14 +432,14 @@ class _StateFile(object):
         """
         Utility class to read and write dill-ed state dictionaries from/to a file into a namespace.
         """
-        def __init__(self, stateDict):
+        def __init__(self, stateDict: Dict[str, Any]):
             assert isinstance(stateDict, dict)
             self.__dict__.update(stateDict)
 
         @classmethod
         @abstractmethod
         @contextmanager
-        def open(cls, outer=None):
+        def open(cls, outer: Optional[Any] = None) -> Iterator[Any]:
```

Comment:
I was thinking of using `Any` here since outer can be an instance of any calling class and the value returned by the Iterator would depend on that calling class.